parsec/backend/postgresql/pki.py

```diff
@@ -0,0 +1,439 @@
+                    submit_payload_signature=entry["submit_payload_signature"],
+                    submit_payload=entry["submit_payload"],
+                )
+                for entry in entries
+            ]
+
+    async def reject(
+        self, organization_id: OrganizationID, enrollment_id: UUID, rejected_on: DateTime
+    ) -> None:
+        """
+        Raises:
+            PkiEnrollmentNotFoundError
+            PkiEnrollmentNoLongerAvailableError
+        """
+
+        async with self.dbh.pool.acquire() as conn, conn.transaction():
+            row = await conn.fetchrow(
+                *_q_get_pki_enrollment_for_update(
+                    organization_id=organization_id.str, enrollment_id=enrollment_id
+                )
+            )
+            if not row:
+                raise PkiEnrollmentNotFoundError()
+            if row["enrollment_state"] != PkiEnrollmentStatus.SUBMITTED.value:
+                raise PkiEnrollmentNoLongerAvailableError()
+
+            await conn.execute(
+                *_q_reject_pki_enrollment(
+                    organization_id=organization_id.str,
+                    enrollment_id=enrollment_id,
+                    enrollment_state=PkiEnrollmentStatus.REJECTED.value,
+                    rejected_on=rejected_on,
+                )
+            )
+            await send_signal(
+                conn, BackendEvent.PKI_ENROLLMENT_UPDATED, organization_id=organization_id
+            )
+
+    async def accept(
+        self,
+        organization_id: OrganizationID,
+        enrollment_id: UUID,
+        accepter_der_x509_certificate: bytes,
+        accept_payload_signature: bytes,
+        accept_payload: bytes,
+        accepted_on: DateTime,
+        user: User,
+        first_device: Device,
+    ) -> None:
+        """
+        Raises:
+            PkiEnrollmentNotFoundError
+            PkiEnrollmentNoLongerAvailableError
+            PkiEnrollmentAlreadyExistError
+            PkiEnrollmentActiveUsersLimitReached
+        """
+        async with self.dbh.pool.acquire() as conn, conn.transaction():
+            row = await conn.fetchrow(
+                *_q_get_pki_enrollment_for_update(
+                    organization_id=organization_id.str, enrollment_id=enrollment_id
+                )
+            )
+            if not row:
+                raise PkiEnrollmentNotFoundError()
+            if row["enrollment_state"] != PkiEnrollmentStatus.SUBMITTED.value:
+                raise PkiEnrollmentNoLongerAvailableError()
+
+            try:
+                await q_create_user(
+                    conn=conn, organization_id=organization_id, user=user, first_device=first_device
+                )
+
+            except UserAlreadyExistsError as exc:
+                raise PkiEnrollmentAlreadyExistError from exc
+
+            except UserActiveUsersLimitReached as exc:
+                raise PkiEnrollmentActiveUsersLimitReached from exc
+
+            assert user.user_certifier is not None
```

Comment:
Add this comment `#  Only the user created during organization bootstrap has no certifier`